client/src/components/app.tsx

```diff
@@ -22,22 +22,36 @@ import { selectIsSeamlessEnabled } from "../selectors/datasetMetadata";
 import Graph from "./graph/graph";
 import GeneInfo from "./geneExpression/geneInfo/geneInfo";
 import Scatterplot from "./scatterplot/scatterplot";
-import EmbeddingPanel from "./EmbeddingPanel";
+import PanelEmbedding from "./PanelEmbedding";
 
 interface StateProps {
-  loading: boolean;
-  error: string;
+  loading: RootState["controls"]["loading"];
+  error: RootState["controls"]["error"];
   graphRenderCounter: number;
   tosURL: string | undefined;
-  privacyURL: string | undefined;
+  privacyURL: string;
   seamlessEnabled: boolean;
-  datasetMetadataError: string | null;
-  isCellGuideCxg: boolean;
-  scatterplotXXaccessor: string;
-  scatterplotYYaccessor: string;
+  datasetMetadataError: RootState["datasetMetadata"]["error"];
+  isCellGuideCxg: RootState["controls"]["isCellGuideCxg"];
+  scatterplotXXaccessor: RootState["controls"]["scatterplotXXaccessor"];
+  scatterplotYYaccessor: RootState["controls"]["scatterplotYYaccessor"];
```

Comment:
no need to redefine the redux state attirbute types, yay!

Reply:
non-blocking: would be nice to extract this to a separate types file